contents/newsletter/choosing-technologies.md

```diff
@@ -0,0 +1,115 @@
+- Based on your evaluation, which one do you recommend?
+
+This isn't just a box to tick, it is a core part of helping us choose the right technologies. For example, our RFC for adopting [Temporal at PostHog](https://github.com/PostHog/meta/pull/99) had 2,594 words, roughly 1/4 of the company as reviewers, 46 comments, and quite a bit of disagreement before finally being adopted. It's now happily in use for [batch exports](/blog/temporal-exports) and [warehouse syncs](/docs/data-warehouse/setup).
+
+Whatever happens, the final decision usually rests on the team implementing and maintaining the technology. Even if an agreement isn't reached, if they feel confident it is the right solution, they have leeway to go ahead with adoption.
+
+> **Related:** [How we work asynchronously](https://newsletter.posthog.com/p/how-we-work-asynchronously)
+
+## When do we say NO to technologies?
+
+If someone goes through all of this work, the expected answer for adopting a technology is yes. What has happened in cases where it ended up as a no?
+
+- **Different perspectives on the evaluation criteria.** For example, the benefit of adopting a new technology for another team is not as valuable as expected, ruining the overall benefit.
+
+- **Additional criteria are added and judged as more important.** For example, features needed for another team to adopt the technology are missing.
+
+- **The evaluation goes against our expectations significantly.** For example, a test ends up being much more significant than expected.
+
+An example of this last one is our decision not to adopt Amazon's Elastic File System (EFS). This was evaluated as a solution to our session replay service rebalancing problems. It needed to re-process historic data when rebalancing, which slowed throughput dramatically, and caused many sleepless nights for the team.
+
+In our test of EFS with real data, rebalances had little-to-no affect on throughput, smaller warmup costs, and other benefits. It seemed to be the solution to our problems, until the cost graphs caught up. We initially estimated $300/month, but this was based on an average throughput, not total. The actual cost was actually **$600/day**. The team quickly concluded that this was an important problem, but the cost was too much. They needed to figure out another solution.
+
+![Expensive EFS](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_10_29_at_09_35_49_930fe4af2d.png)
+
+## Continuously evaluating
+
+Finally choosing a technology does not mean you are done. Technologies have ongoing costs in terms of maintenance, support, recruitment, speed, and more. Why shouldn't your evaluation be ongoing also?
+
+We regularly evaluate even the most cherished of our technologies to ensure they are the right choice. For example, ClickHouse is core to what we do at PostHog. We have long been big advocates of ClickHouse, but that doesn't stop us from considering and testing other tools, such as ByConity and Apache Iceberg.
```

Comment:
I'd like to hear more about the ByConity / Iceberg conversations more. What have we been evaluating, have we made any decisions and if so, why?